# 
# patch "ns-2-snapshot-20050118/classifier/classifier-addr.h"
#  from [403dc0bbf9252ac0f8fd8952a12a1fc444549618]
#    to [1e33f9d61dab6e2004cbf9d273b5b5d8a323bd8c]
# 
# patch "ns-2-snapshot-20050118/classifier/classifier-hash.cc"
#  from [4a7b6b839dc9c3ad0b3f216adab5e78d005e335b]
#    to [6f9ec72b0c324039f6d2005f4eb29ed0d1007ad6]
# 
# patch "ns-2-snapshot-20050118/classifier/classifier-port.h"
#  from [7c49605694507823d1c727caa8a00a8d7b2772c4]
#    to [07e04257794e7bde8142c34a1a91bfd44ab9f733]
# 
# patch "ns-2-snapshot-20050118/classifier/classifier.cc"
#  from [a6bf9a643f5ce2890ff9a78f19fadeacf1385735]
#    to [6cd44a781ff49328648e3646e3713fd1aef5d16a]
# 
# patch "ns-2-snapshot-20050118/classifier/classifier.h"
#  from [4cfe252236e1bd376ee0e31a858d8a9a31c92e9c]
#    to [5b018a9b97cf50e1473974b3df1a21b0b7b7ffdf]
# 
--- ns-2-snapshot-20050118/classifier/classifier-addr.h
+++ ns-2-snapshot-20050118/classifier/classifier-addr.h
@@ -46,10 +46,11 @@
 class ReserveAddressClassifier : public AddressClassifier {
 public:
 	ReserveAddressClassifier() : AddressClassifier(), reserved_(0) {}
+ private:
+	virtual int getnxt(NsObject *);
 protected:
 	virtual int classify(Packet *p);
 	virtual void clear(int slot);
-	virtual int getnxt(NsObject *);
 	virtual int command(int argc, const char*const* argv);
 	int reserved_;
 };
--- ns-2-snapshot-20050118/classifier/classifier-hash.cc
+++ ns-2-snapshot-20050118/classifier/classifier-hash.cc
@@ -191,7 +191,7 @@
 
 void DestHashClassifier::do_install(char* dst, NsObject *target) {
 	nsaddr_t d = atoi(dst);
-	int slot = getnxt(target);
+	int slot = allocPort (target);
 	install(slot, target); 
 	if (set_hash(0, d, 0, slot) < 0)
 		fprintf(stderr, "DestHashClassifier::set_hash from within DestHashClassifier::do_install returned value < 0");
--- ns-2-snapshot-20050118/classifier/classifier-port.h
+++ ns-2-snapshot-20050118/classifier/classifier-port.h
@@ -40,9 +40,10 @@
 class ReservePortClassifier : public PortClassifier {
 public:
 	ReservePortClassifier() : PortClassifier(), reserved_(0) {}
+private:
+	virtual int getnxt(NsObject *);
 protected:
 	void clear(int slot);
-	int getnxt(NsObject *);
 	int command(int argc, const char*const* argv);
 	int reserved_;
 };
--- ns-2-snapshot-20050118/classifier/classifier.cc
+++ ns-2-snapshot-20050118/classifier/classifier.cc
@@ -117,6 +117,11 @@
 	}
 }
 
+int Classifier::allocPort (NsObject *nullagent)
+{
+	return getnxt (nullagent);
+}
+
 int Classifier::getnxt(NsObject *nullagent)
 {
 	int i;
--- ns-2-snapshot-20050118/classifier/classifier.h
+++ ns-2-snapshot-20050118/classifier/classifier.h
@@ -74,8 +74,10 @@
 	// hierarchical specific
 	virtual void set_table_size(int level, int nn) {}
 	
-protected:
+	int allocPort (NsObject *);
+private:
 	virtual int getnxt(NsObject *);
+protected:
 	virtual int command(int argc, const char*const* argv);
 	void alloc(int);
 	NsObject** slot_;	/* table that maps slot number to a NsObject */
